WcaOnRails/app/controllers/persons_controller.rb

```diff
@@ -25,4 +25,17 @@ def index
       end
     end
   end
+
+  def show
+    @person = Person.current.includes(:user, :ranksSingle, :ranksAverage, :competitions).find_by_wca_id(params[:id])
+    @previous_persons = Person.where(wca_id: params[:id]).where.not(subId: 1).order(:subId)
+    @ranks_single = @person.ranksSingle
+    @ranks_average = @person.ranksAverage
+    @events_competed_in = Event.where(id: (@ranks_single.map(&:eventId) + @ranks_average.map(&:eventId)).uniq)
```

Comment:
Good call. I'm gonna add an `event` method to the `Ranks*` classes using the `c_find` and then do the following:
```ruby
@events_competed_in = (@ranks_single.map(&:event) & @ranks_average.map(&:event)).sort_by(&:rank)
```